index_impl.go

```diff
@@ -482,6 +483,14 @@ func (i *indexImpl) SearchInContext(ctx context.Context, req *SearchRequest) (sr
 	ctx = context.WithValue(ctx, search.SearchIOStatsCallbackKey,
 		search.SearchIOStatsCallbackFunc(sendBytesRead))
 
+	bufPool := s2.NewGeoBufferPool(24 * 1024, 24)
+	getBufferPool := func() *s2.GeoBufferPool {
+		return bufPool
```

Comment:
For queries which doesn't have geoshape component in it, we shouldn't call NewGeoBufferPool.
To avoid that, 

```
var geoBufPool *s2.GeoBufferPool

getBufferPool := func() *s2.GeoBufferPool {
  if geoBufPool == nil {
    geoBufPool = s2.NewGeoBufferPool(MinBufPoolSize, MaxBufPoolSize)
  }
  return geoBufPool
}
```